libcontainer/cgroups/fs/paths.go

```diff
@@ -122,13 +122,11 @@ func getCgroupData(c *configs.Cgroup, pid int) (*cgroupData, error) {
 		return nil, errors.New("cgroup: either Path or Name and Parent should be used")
 	}
 
-	// XXX: Do not remove this code. Path safety is important! -- cyphar
-	cgPath := utils.CleanPath(c.Path)
-	cgParent := utils.CleanPath(c.Parent)
-	cgName := utils.CleanPath(c.Name)
-
-	innerPath := cgPath
+	// XXX: Do not remove CleanPath. Path safety is important! -- cyphar
+	innerPath := utils.CleanPath(c.Path)
 	if innerPath == "" {
+		cgParent := utils.CleanPath(c.Parent)
+		cgName := utils.CleanPath(c.Name)
 		innerPath = filepath.Join(cgParent, cgName)
```

Comment:
The "clean" part is a bit more explicit the non-inlined way. I think this is why it was written the way it was -- to make it more explicit that we're cleaning all the input.

I guess the compiler optimizes this out anyway.

Reply:
Yeah the angry comment was added because `CleanPath` was deleted by someone inadvertently during a cleanup which reintroduced a vulnerability, so I tried to make it as clear as possible that the `CleanPath` is necessary and must not be removed.